internal/api/vm/converters.go

```diff
@@ -0,0 +1,71 @@
+package vm
+
+import (
+	"github.com/kyma-project/btp-manager/internal/service-manager/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func ToSecretVM(list v1.SecretList) Secrets {
+	secrets := Secrets{
+		Items: []Secret{},
+	}
+	for _, secret := range list.Items {
+		secrets.Items = append(
+			secrets.Items, Secret{
+				Name:      secret.Name,
+				Namespace: secret.Namespace,
+			},
+		)
+	}
+	return secrets
+}
+
+func ToServiceOfferingsVM(offerings *types.ServiceOfferings) ServiceOfferings {
+	serviceOfferings := ServiceOfferings{
+		NumItems: len(offerings.ServiceOfferings),
+		Items:    []ServiceOffering{},
+	}
+
+	for _, offering := range offerings.ServiceOfferings {
+		imageUrl, _ := offering.MetadataValueByFieldName(types.ServiceOfferingImageUrl)
+		displayName, _ := offering.MetadataValueByFieldName(types.ServiceOfferingDisplayName)
+		offering := ServiceOffering{
+			ID:          offering.ID,
+			Description: offering.Description,
+			CatalogID:   offering.CatalogID,
+			CatalogName: offering.CatalogName,
+			Metadata: ServiceOfferingMetadata{
+				ImageUrl:    imageUrl,
+				DisplayName: displayName,
+			},
+		}
+		serviceOfferings.Items = append(serviceOfferings.Items, offering)
+	}
+	return serviceOfferings
+}
+
+func ToServiceInstancesVM(serviceInstances []unstructured.Unstructured) ServiceInstances {
```

Comment:
Service Instances will be provided by SM client in an appropriate struct then func will need adjustments. I believe it's better to wait for SM client functionality than work on unstructured objects.